### PR TITLE
[OneExplorer] Remove 'oneTree' member

### PR DESCRIPTION
**Pull request by dayo09**
Let's remove 'oneTree' member.
**This change is necessary for a bug fix for refresh command.**
NOTE that OneNodes are generated on the first expansion of tree view.
'oneTree' is not used as getChildren method should returns
freshly generated trees.

ONE-vscode-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

### CODE REVIEW

File: src/OneExplorer.ts [hunk 1]
```diff
@@ -77,15 +71,15 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
...
-    if (this.oneTree === undefined) {
-      vscode.window.showInformationMessage('No ONE model or config in empty workspace');
-      return Promise.resolve([]);
-    }
-
     if (element) {
       return Promise.resolve(this.getNode(element.node));
     } else {
-      return Promise.resolve(this.getNode(this.oneTree));
+      if (!this.workspaceRoot) {
```
Comment (jyoungyun) [inline src/OneExplorer.ts:77]: How about locating this function outside this `if-else` phrase?
This change does not change the operation, but it's easier to understand the code.

Note.
If the `this.workspaceRoot` is undefined, element is always undefined.

**Review by jyoungyun — APPROVED** [review]
LGTM Thank you!

File: src/OneExplorer.ts [hunk 2]
```diff
@@ -77,15 +71,15 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
   }
 
   getChildren(element?: OneNode): OneNode[]|Thenable<OneNode[]> {
-    if (this.oneTree === undefined) {
-      vscode.window.showInformationMessage('No ONE model or config in empty workspace');
+    if (!this.workspaceRoot) {
+      vscode.window.showInformationMessage('Cannot find workspace root');
       return Promise.resolve([]);
```
Comment (hyunsik-yoon) [inline src/OneExplorer.ts:76]: (not a part of this PR) but would `Promise.reject()` be correct?

**Comment by hyunsik-yoon** [comment]
I wonder if the approach would fast enough if user has many files, e.g., about 20 models and 4 cfg files per model.

**Comment by dayo09** [comment]
@hyunsik-yoon It doens't change the performance. The number of tree-search operator is the same. The original code makes bug on refresh command(refresh not working).

The child nodes are generated on the first whole-tree generation(getTree). 
getChildren is called when you first expand the node and use the nodes generated on getTree. (it converts nodes into oneNode)
Refresh command expects that getChildren with an empty argument to get the fresh tree, but it just remembers oneTree before. This commit fixes it.

**Comment by jyoungyun** [comment]
> I wonder if the approach would fast enough if user has many files, e.g., about 20 models and 4 cfg files per model.

It takes some minutes at the first loading time, but once it is loaded, there will no issues of speed until pressing Refresh button.
Modules that take longer time to load in the first loading time of other extension modules are showing blue progress bar. And I think this operation will work similarly in our Explorer view.

File: src/OneExplorer.ts (hunk 2 above)
Comment (dayo09) [reply src/OneExplorer.ts:76]: image

Comment (jyoungyun) [reply src/OneExplorer.ts:76]: Please refer to this https://github.com/microsoft/vscode-extension-samples/blob/main/tree-view-sample/src/nodeDependencies.ts#L22-L25 codes.

I think we don't need to show error message. Instead we can show the information noti for this one.

**Comment by hyunsik-yoon** [comment]
@dayo09 @jyoungyun Thanks for explanation about `getChildren()`. 😸

**Review by hyunsik-yoon — APPROVED** [review]

File: src/OneExplorer.ts (hunk 2 above)
Comment (dayo09) [reply src/OneExplorer.ts:76]: Yes, I took the same reference. Both cases looks fine. You can choose. Or maybe we can discuss another time.

**Pull request merged**